evalml/pipelines/utils.py

```diff
@@ -637,8 +639,22 @@ def generate_pipeline_code(element):
             element.__class__.__name__,
         ),
     )
+    has_dfs = "DFS Transformer" in element.component_graph.compute_order
+    if has_dfs and not features_path:
+        warnings.warn(
+            "This pipeline contains a DFS Transformer but no `features_path` has been specified. Please add a `features_path` or the pipeline code will generate a pipeline that does not run DFS.",
+        )
```

Comment:
What's our thought/process on using `warnings.warn` vs `logger.warning`? Do we have one? A quick search of the code shows instances of using both.

Reply:
```
[warnings.warn()](https://docs.python.org/3/library/warnings.html#warnings.warn) in library code if the issue is avoidable and the client application should be modified to eliminate the warning

[logging.warning()](https://docs.python.org/3/library/logging.html#logging.warning) if there is nothing the client application can do about the situation, but the event should still be noted
```
https://docs.python.org/3/howto/logging.html#when-to-use-logging

I believe this matches the former!